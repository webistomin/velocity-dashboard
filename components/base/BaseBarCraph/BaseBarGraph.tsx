import { VueComponent } from 'types/vue-components';
import { Component, Prop, mixins } from 'nuxt-property-decorator';
import { ChartData, ChartOptions } from 'chart.js';
import { Bar } from 'vue-chartjs';

export interface IBaseBarGraphProps {
  chartData: ChartData;
  options?: ChartOptions;
  fallbackText?: string;
}

@Component({
  name: 'BaseBarGraphWithProps',
})
class BaseBarGraphWithProps extends VueComponent<IBaseBarGraphProps> {
  @Prop()
  public readonly chartData!: IBaseBarGraphProps['chartData'];

  @Prop()
  public readonly options!: IBaseBarGraphProps['options'];

  @Prop({ default: 'Bar graph' })
  public readonly fallbackText: IBaseBarGraphProps['fallbackText'];
}

@Component({
  name: 'BaseBarGraph',
})
export default class BaseBarGraph extends mixins(Bar, BaseBarGraphWithProps) {
  mounted(): void {
    this.renderChart(this.chartData, this.options);

    this.$nextTick(() => {
      const canvas = this.$refs.canvas as HTMLCanvasElement;
      canvas.setAttribute('aria-label', this.fallbackText || 'Bar graph');
      canvas.setAttribute('role', 'img');
    });
  }
}
