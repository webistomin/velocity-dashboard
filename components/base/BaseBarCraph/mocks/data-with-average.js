export const barGraphDataWithAverage = {
  labels: ['Mon', 'Tue', 'Fri', 'Thu', 'Wed', 'Sat', 'Sun'],
  datasets: [
    {
      label: 'Average',
      borderColor: '#F7C137',
      fill: false,
      lineTension: 0,
      data: [110, 80, 100, 110, 74, 55, 124],
      type: 'line',
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      backgroundColor: '#2E5BFF',
      data: [82, 65, 75, 130, 55, 62, 120],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [90, 75, 80, 60, 45, 117],
    },
  ],
};
