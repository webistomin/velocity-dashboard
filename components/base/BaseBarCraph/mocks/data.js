export const barGraphData = {
  labels: ['April', 'May', 'June'],
  datasets: [
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      lineTension: 0.4,
      backgroundColor: '#2E5BFF',
      data: [60, 65, 85],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [70, 80, 90],
    },
  ],
};
