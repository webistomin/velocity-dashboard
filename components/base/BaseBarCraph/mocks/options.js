export const options = {
  responsive: true,
  maintainAspectRatio: false,
  scales: {
    yAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          suggestedMin: 0,
          suggestedMax: 125,
          fontColor: '#B0BAC9',
        },
      },
    ],
    xAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          fontColor: '#B0BAC9',
        },
      },
    ],
  },
  legend: {
    display: true,
    align: 'end',
    labels: {
      boxWidth: 10,
      usePointStyle: true,
      padding: 20,
      fontColor: '#B0BAC9',
      fontFamily: 'Rubik, system-ui, sans-serif',
    },
  },
  elements: {
    point: {
      radius: 5,
      hoverRadius: 4,
      borderWidth: 3,
      hoverBorderWidth: 3,
      backgroundColor: '#FFF',
    },
  },
};
