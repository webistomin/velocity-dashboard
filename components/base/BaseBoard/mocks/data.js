export const boardData = [
  {
    title: 'Backlog',
    color: 'blue',
    tasks: [
      {
        title: 'Spire1',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire2',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire3',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'In progress',
    color: 'purple',
    tasks: [
      {
        title: 'Spire5',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire6',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire7',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire8',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Validate',
    color: 'darkturquoise',
    tasks: [
      {
        title: 'Spire9',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire10',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire11',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire12',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Complete',
    color: 'yellow',
    tasks: [
      {
        title: 'Spire13',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire14',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire15',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire16',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
];
