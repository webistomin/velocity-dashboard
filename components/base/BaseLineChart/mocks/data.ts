import { ChartData } from 'chart.js';

export const lineGraphData: ChartData = {
  labels: [
    '',
    'Mar 1',
    'Mar 8',
    'Mar 15',
    'Mar 22',
    'Mar 29',
    'Apr 5',
    'Apr 12',
    'Apr 19',
    'Apr 26',
    'May 2',
    'May 9',
    'May 16',
    'May 23',
    'May 30',
    'Jun 5',
    'Jun 10',
    'Jun 19',
  ],
  datasets: [
    {
      label: 'Revenue',
      borderColor: '#2E5BFF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 8000,
        },
        {
          x: 1,
          y: 11500,
        },
        {
          x: 2,
          y: 9800,
        },
        {
          x: 3,
          y: 15100,
        },
        {
          x: 4,
          y: 23000,
        },
        {
          x: 5,
          y: 22000,
        },
        {
          x: 6,
          y: 16000,
        },
        {
          x: 7,
          y: 13800,
        },
        {
          x: 8,
          y: 16300,
        },
        {
          x: 9,
          y: 17000,
        },
        {
          x: 10,
          y: 24500,
        },
        {
          x: 11,
          y: 19000,
        },
        {
          x: 12,
          y: 12000,
        },
        {
          x: 13,
          y: 8900,
        },
        {
          x: 14,
          y: 16000,
        },
        {
          x: 15,
          y: 27000,
        },
        {
          x: 16,
          y: 22000,
        },
        {
          x: 17,
          y: 21000,
        },
      ],
    },
    {
      label: 'Trips',
      borderColor: '#8C54FF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 4800,
        },
        {
          x: 1,
          y: 5000,
        },
        {
          x: 2,
          y: 7500,
        },
        {
          x: 3,
          y: 7000,
        },
        {
          x: 4,
          y: 11500,
        },
        {
          x: 5,
          y: 13000,
        },
        {
          x: 6,
          y: 7500,
        },
        {
          x: 7,
          y: 8500,
        },
        {
          x: 8,
          y: 6000,
        },
        {
          y: 13700,
        },
        {
          x: 10,
          y: 18000,
        },
        {
          x: 11,
          y: 17600,
        },
        {
          x: 12,
          y: 2500,
        },
        {
          x: 13,
          y: 6000,
        },
        {
          x: 14,
          y: 9000,
        },
        {
          x: 15,
          y: 22500,
        },
        {
          x: 16,
          y: 14000,
        },
        {
          x: 17,
          y: 19000,
        },
      ],
    },
  ],
};
