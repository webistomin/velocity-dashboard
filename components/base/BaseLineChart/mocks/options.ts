import { ChartOptions } from 'chart.js';

export const options: ChartOptions = {
  responsive: true,
  maintainAspectRatio: false,
  scales: {
    yAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          suggestedMin: 0,
          suggestedMax: 125,
          fontColor: '#B0BAC9',
          callback: (value) => {
            let tempVal = value.toString();
            if (tempVal.length >= 4) tempVal = tempVal.substring(0, tempVal.length - 3) + 'k ';
            return '$' + tempVal;
          },
        },
      },
    ],
    xAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          fontColor: '#B0BAC9',
        },
      },
    ],
  },
  legend: {
    display: true,
    align: 'end',
    labels: {
      boxWidth: 10,
      usePointStyle: true,
      padding: 20,
      fontColor: '#B0BAC9',
    },
  },
  elements: {
    point: {
      radius: 5,
      hoverRadius: 4,
      borderWidth: 3,
      hoverBorderWidth: 3,
    },
  },
};
