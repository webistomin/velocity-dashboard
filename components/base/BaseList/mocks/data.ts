export const listData = [
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
];
