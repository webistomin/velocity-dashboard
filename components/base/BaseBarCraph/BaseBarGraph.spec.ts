import { mount } from '@vue/test-utils';

import { options } from './mocks/options';
import { barGraphData } from './mocks/data';
import BaseBarGraph from './BaseBarGraph';

describe('BaseBarGraph', () => {
  const factory = (propsData = {}) => {
    return mount(BaseBarGraph, {
      propsData: {
        options,
        chartData: barGraphData,
        ...propsData,
      },
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory();
    const html = wrapper.html();
    expect(html).toMatchSnapshot();
  });

  it('Set default a11y attributes', async () => {
    const wrapper = factory();
    const vm = wrapper.vm;
    await vm.$nextTick();

    const canvas = vm.$refs.canvas as HTMLCanvasElement;
    const role = canvas.getAttribute('role');
    const ariaLabel = canvas.getAttribute('aria-label');

    expect(role).toEqual('img');
    expect(ariaLabel).toEqual('Bar graph');
  });

  it('Set aria-label from props', async () => {
    const expectedAriaLabel = 'My brand new bar graph';
    const wrapper = factory({
      fallbackText: expectedAriaLabel,
    });
    const vm = wrapper.vm;
    await vm.$nextTick();

    const canvas = vm.$refs.canvas as HTMLCanvasElement;
    const ariaLabel = canvas.getAttribute('aria-label');

    expect(ariaLabel).toEqual(expectedAriaLabel);
  });
});
