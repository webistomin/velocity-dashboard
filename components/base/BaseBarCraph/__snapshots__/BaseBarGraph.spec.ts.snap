// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBadge Matches snapshot 1`] = `"<div class=\\"\\"><canvas id=\\"bar-chart\\" width=\\"0\\" height=\\"0\\" style=\\"display: block;\\" class=\\"chartjs-render-monitor\\"></canvas></div>"`;

exports[`BaseBarGraph Matches snapshot 1`] = `"<div class=\\"\\"><canvas id=\\"bar-chart\\" width=\\"0\\" height=\\"0\\" style=\\"display: block;\\" class=\\"chartjs-render-monitor\\"></canvas></div>"`;
